esphome/components/atm90e32/text_sensor/__init__.py

```diff
@@ -0,0 +1,63 @@
+import esphome.codegen as cg
+from esphome.components import text_sensor
+import esphome.config_validation as cv
+from esphome.const import CONF_ID, CONF_PHASE_A, CONF_PHASE_B, CONF_PHASE_C
+
+from .. import atm90e32_ns
+from ..sensor import ATM90E32Component
+
+ATM90E32PhaseStatusSensor = atm90e32_ns.class_(
+    "ATM90E32PhaseStatusSensor",
+    text_sensor.TextSensor,
+    cg.Parented.template(ATM90E32Component),
+)
+
+ATM90E32FreqStatusSensor = atm90e32_ns.class_(
+    "ATM90E32FreqStatusSensor",
+    text_sensor.TextSensor,
+    cg.Parented.template(ATM90E32Component),
+)
+
+CONF_PHASE_STATUS = "phase_status"
+CONF_FREQUENCY_STATUS = "frequency_status"
+PHASE_KEYS = [CONF_PHASE_A, CONF_PHASE_B, CONF_PHASE_C]
+
+PHASE_STATUS_SCHEMA = cv.Schema(
+    {
+        cv.Optional(CONF_PHASE_A): text_sensor.text_sensor_schema(
+            ATM90E32PhaseStatusSensor, icon="mdi:flash-alert"
+        ),
+        cv.Optional(CONF_PHASE_B): text_sensor.text_sensor_schema(
+            ATM90E32PhaseStatusSensor, icon="mdi:flash-alert"
+        ),
+        cv.Optional(CONF_PHASE_C): text_sensor.text_sensor_schema(
+            ATM90E32PhaseStatusSensor, icon="mdi:flash-alert"
+        ),
+    }
+)
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_ID): cv.use_id(ATM90E32Component),
```

Comment:
This way it can be optional if there is only one main sensor entry
```suggestion
        cv.GenerateID(): cv.use_id(ATM90E32Component),
```